jerry-core/jerry.cpp

```diff
@@ -899,6 +900,57 @@ bool jerry_api_get_object_field_value (jerry_api_object_t *object_p,
 }
 
 /**
+ * Applies the given function to the every fields in the objects
+ *
+ */
+bool
+jerry_api_foreach_object_field (jerry_api_object_t *object_p, /**< object */
+                               jerry_object_field_foreach_t foreach_p, /**< foreach function */
+                               void *user_data) /**< user data for foreach function */
+{
+  jerry_assert_api_available ();
+
+  bool continuous = true;
+  for (ecma_property_t *property_p = ecma_get_property_list (object_p);
+       property_p != NULL && continuous;
+       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  {
+    ecma_string_t *property_name_p;
+    if (property_p->type == ECMA_PROPERTY_NAMEDDATA)
+    {
+      property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t, property_p->u.named_data_property.name_p);
+    }
+    else if (property_p->type == ECMA_PROPERTY_NAMEDACCESSOR)
+    {
+      property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t, property_p->u.named_accessor_property.name_p);
+    }
+    else
+    {
+      continue;
```

Comment:
Could you, please, add assertion that `property_p->type == ECMA_PROPERTY_INTERNAL`?